src/sdt.rs

```diff
@@ -52,7 +53,7 @@ impl SdtHeader
         }
 
         // Check that the lowest byte is 0
-        if sum & 0b1111_1111 != 0
+        if sum % 0b1111_1111 != 0
```

Comment:
I'm probably just having a brain-fart, but why this change?

Reply:
It's not clear for FADT in the docs but as for RSDT: All bytes of the table summed must be equal to 0 (mod 0xFF). And since sum is usize it will grow beyond 0xFF and if you do a bitwise and on it, it will not be the same result as if you would use the mod. Another solution would be to use a u8 but not sure what happens when you grow over b1111_1111. Not sure if I was clear

Reply:
Huh, did you see a section in the spec about mod? I was basing my implementation on [here](https://wiki.osdev.org/RSDP) where it says "you add up every byte in the structure and make sure the lowest byte of the result is equal to zero.". Looking around, [Haiku also appears to use `&`](https://github.com/haiku/haiku/blob/master/src/system/boot/platform/efi/acpi.cpp#L50).

To be honest tho, I'm not sure I understand the difference enough to be sure. I'll have a think

Reply:
Okay, so maybe RSDT and RSDP do not have the same way of checking. For RSDT see [here](https://wiki.osdev.org/RSDT#Checksum).

Reply:
And regarding the implementation of Haiku, they use a `unsigned char` which is only 8 bits, so when the reach the max value it goes back to 0

Reply:
So yeah I think we should also use `%` in the RSDP

Reply:
Yup, that sounds right. I don't think using the equivalent in Rust (using `u8` and letting it go back to 0) would even work because it would just panic on the overflow, so let's move to using `%`